### PR TITLE
Bug Fix - Adding an allowed address pair wipes port security groups

### LINKED ISSUE #51755: OpenStack (Cloud Routes): Adding an allowed address pair wipes port security groups
/kind bug

When using the OpensStack cloud provider with cloud routes enabled instances will have their security groups removed when the allowed address pair is added to the instance's port.

Reproduce:
Run `kube-apiserver`, `kube-controller-manager`, and `kubelet` on OpenStack. Enable provisioning cloud routes in `kube-controller-manager` and configure the cloud config appropriately. The instance should have its security group removed when the allowed address pair is added.

This is due to a bug in gophercloud that has since been patched (See gophercloud/gophercloud#509).

To be honest I'm not sure how kubernetes handles scenarios like this (Workaround in code, update library, document manual workaround, etc), so I reported it just in case. Feel free to close this if it's in the wrong place/irrelevant.

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

Fix for cloud routes enabled instances will have their security groups
removed when the allowed address pair is added to the instance's port.

Upstream bug report is in:
https://github.com/gophercloud/gophercloud/issues/509

Upstream bug fix is in:
https://github.com/gophercloud/gophercloud/pull/510

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Fixes #51755

**Special notes for your reviewer**:

Just an fix in vendored code. minimal changes needed in OpenStack cloud provider

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go [hunk 1]
```diff
@@ -994,7 +994,7 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *v1.Serv
 		}
 
 		portID := loadbalancer.VipPortID
-		update_opts := neutronports.UpdateOpts{SecurityGroups: []string{lbSecGroup.ID}}
+		update_opts := neutronports.UpdateOpts{SecurityGroups: &[]string{lbSecGroup.ID}}
```
Comment (FengyunPan) [inline pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: Nice changes.

File: vendor/github.com/gophercloud/gophercloud/openstack/compute/v2/images/doc.go [hunk 2]
```diff
@@ -1,7 +1,32 @@
-// Package images provides information and interaction with the image API
-// resource in the OpenStack Compute service.
-//
-// An image is a collection of files used to create or rebuild a server.
-// Operators provide a number of pre-built OS images by default. You may also
-// create custom images from cloud servers you have launched.
+/*
```
Comment (FengyunPan) [inline vendor/github.com/gophercloud/gophercloud/openstack/compute/v2/images/doc.go]: I don't understand why you introduce all modules of gophercloud? The images and other modules are useless for kubernetes, can we not introduce them?

**Review by FengyunPan — COMMENTED** [review]
@dims Sorry for the delayed review.

File: vendor/github.com/gophercloud/gophercloud/openstack/compute/v2/images/doc.go (hunk 2 above)
Comment (dims) [reply vendor/github.com/gophercloud/gophercloud/openstack/compute/v2/images/doc.go]: @FengyunPan - I had to add it because verify fails otherwise.

**Comment by dims** [comment]
@FengyunPan please approve so i can request the release team to add v1.8 milestone

File: vendor/github.com/gophercloud/gophercloud/openstack/compute/v2/images/doc.go (hunk 2 above)
Comment (FengyunPan) [reply vendor/github.com/gophercloud/gophercloud/openstack/compute/v2/images/doc.go]: All right.

**Comment by FengyunPan** [comment]
/lgtm
@dims I can only add lgtm label, I don't have permission (I am not a approver) to add approve label, so @anguslees PTAL, thank you.

**Comment by dims** [comment]
@jdumars @calebamiles Can you please add the 1.8 milestone flag for this bug-fix PR?

@brendanburns @dchen1107 Can you please approve this PR? It's mostly vendor code update and minor change to the code in our repo (already ok'ed by a long time contributor)

**Comment by dims** [comment]
@FengyunPan we can ignore the kops test for now. it's a known problem

**Comment by dims** [comment]
@kubernetes/sig-release-members Can you please add the 1.8 milestone flag for this bug-fix PR?

**Comment by calebamiles** [comment]
Added the PR back to the milestone, @dims.

**Pull request merged**